### PR TITLE
Modified error message order in `InvalidParameterException`

### PR DESCRIPTION
**Pull request by Aritra8438**
# Description

Modify the function to show the error in format `parameter: error_message`

image

image

Fixes: #1113 

# Comments

- Changed the order of the affected parameter and the error.
- Used try-catch block to throw more specific errors.

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [x] If you are creating a feature, the new tests were added.
- [x] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

### CODE REVIEW

**Comment by Aritra8438** [comment]
Hi @pabo99, just a friendly ping here too!

**Comment by pabo99** [comment]
> Hi @pabo99, just a friendly ping here too!

I believe the error message is still confusing. The best way to address this issue is to revisit the changes (and any remaining suggestions) from PR #7435.

**Comment by Aritra8438** [comment]
Hi @pabo99, I was going through the codebase and the referred PR, I will try to make the error more informative, but not too sure if I will be able to make it less confusing (or more readable).

And how do I run the Python or PHP tests until my first PR gets merged (then I will be able to run tests from the github workflow)?

**Comment by pabo99** [comment]
> Hi @pabo99, I was going through the codebase and the referred PR, I will try to make the error more informative, but not too sure if I will be able to make it less confusing (or more readable) because of `additional parameters`.
> 
> And how do I run the Python or PHP tests until my first PR gets merged (then I will be able to run tests from the github workflow)?

This is the command to run PHP tests: `./stuff/runtests.sh` . 

Python tests are not as necessary in these cases, as most of those tests are for cronjobs; whatever, the command to run them is as follows: `python3 -m pytest ./stuff/`

You must run these commands inside the Docker container.

**Comment by Aritra8438** [comment]
Done @pabo99, mostly followed the comments of the referred PR and your comments.
PTAL!

File: frontend/tests/controllers/ProblemCreateTest.php [hunk 1]
```diff
@@ -120,7 +120,7 @@ public function testCreateWithInvalidAlias(string $alias) {
             \OmegaUp\Controllers\Problem::apiCreate($r);
             $this->fail('Problem creation should have failed');
         } catch (\OmegaUp\Exceptions\InvalidParameterException $e) {
-            $this->assertSame($e->getMessage(), 'parameterInvalid');
+            $this->assertSame($e->getMessage(), 'parameterStringTooLong');
```
Comment (pabo99) [inline frontend/tests/controllers/ProblemCreateTest.php:123]: It seems we resolve other hidden issue. This test is asserting that all the data provided throws an error. But now we are detecting one of these cases throws a different message, which is true.

You need fix this function and the data provider function as well.

The return value of `invalidAliasValueProvider` function should include the expected message:
```php
        return [
            ['this has a space', 'parameterInvalid'],
            ['this-alias-is-way-too-long-and-should-be-rejected', 'parameterStringTooLong'],
            ['colons:are-disallowed', 'parameterInvalid'],
            ['new', 'parameterInvalid'],  // restricted alias
        ];
```
Then, this function will receive this new parameter: `$expectedMessage` (the type is string).
And finally you can use this new parameter here:
```php
            $this->assertSame($e->getMessage(), $expectedMessage);
```

**Review by pabo99 — COMMENTED** [review]
One last comment.

Don't forget to run the command `./stuff/lint.sh` before pushing your changes.

File: frontend/tests/controllers/ProblemCreateTest.php (hunk 1 above)
Comment (Aritra8438) [reply frontend/tests/controllers/ProblemCreateTest.php:123]: Sure @pabo99, I tried to fix the test like this:
```php
$this->assertSame(
                true,
                in_array(
                    $e->getMessage(),
                    ['parameterStringTooLong', 'parameterInvalid']
                )
            );
```

> The return value of `invalidAliasValueProvider` function should include the expected message:

Thanks for this, looks great.

Comment (pabo99) [reply frontend/tests/controllers/ProblemCreateTest.php:123]: Your approach will fix the error in the test, but doesn't guarantee the correct message for each case. It's better to fix it from the dataProvider.

Comment (Aritra8438) [reply frontend/tests/controllers/ProblemCreateTest.php]: Done! @pabo99.

**Review by pabo99 — APPROVED** [review]
LGTM!

**Review by heduenas — APPROVED** [review]

**Pull request merged**